routes/api/v1/favorites.js

```diff
@@ -0,0 +1,37 @@
+var express = require("express");
+var fetch = require("node-fetch");
+var router = express.Router();
+var user = require('../../../models').User
+var favorite = require('../../../models').Favorite
+
+router.post("/", function(req, res, next) {
+  res.setHeader("Content-Type", "application/json");
+  let city = req.body.location
+
+  if (!req.body.apiKey) {
+    return res.status(401).send({ error: "Missing API Key" });
+  }
+
+  user.findOne({
+    where: { apiKey: req.body.apiKey }
+  })
+  .then(user => {
+    if (!user) {
+      return res.status(401).send({ error: "Missing API Key" });
+    }
+      return user.createFavorite({
+        location: city
+      })
+      .then(favorite => {
+        res.status(200).send({
+           message: `${city} has been added to your favorites`
+         })
+      })
+    .catch(error => {
+      res.status(500).send({ error: 'favorite not created' });
```

Comment:
It might be a good idea to separate this error out into multiple cases. For example, it would be more clarifying for someone to receive an error that would tell them that perhaps they didn't send a valid location in the body. Or their API key wasn't valid

Reply:
Thanks for the feedback Tay!  I've updated this request so that my error messages are more clear and I am checking for a missing location as well as api key.

Reply:
Thanks for the feedback Tay!  I've updated this request so that my error messages are more clear and I am checking for a missing location as well as api key.